tensorflow/compiler/xla/service/hlo_graph_dumper.h

```diff
@@ -63,8 +63,13 @@ string DumpGraph(const HloComputation& computation, const string& label,
 // The number of nodes dumped is controlled by the radius parameter, which
 // (roughly) corresponds to the max distance a node may be from the primary node
 // before it's omitted from the graph.
-string DumpNeighborhoodAround(const HloInstruction& node, int radius,
-                              bool show_backend_config = false);
+//
+// The optional boundary specifies a set of boundary nodes, beyond which nodes
+// will be omitted even if they are within the radius.
+string DumpNeighborhoodAround(
+    const HloInstruction& node, int radius,
+    const absl::flat_hash_set<const HloInstruction*>& boundary = {},
```

Comment:
New default parameter should be moved to the end of the parameter list so it does not break existing callers that may pass `show_backend_config`.

Reply:
I was not sure whether 'boundary' or 'show_backend_config' would be more likely to get a default value. For my use case, show_backend_config is more often to be false, and I did not find any other case of DumpNeighborhoodAround() in the code base. But you have a good point. I will update the PR.

Reply:
No need to. I've made the necessary change during internal patch review. This pull request should land soon.

Reply:
Looks like our email crossed. Please review the new PR, which also addresses the missing ''' issue in the error message that Justin mentioned.